example/simple/feature/main/src/main/java/com/freeletics/mad/example/feature/main/MainDI.kt

```diff
@@ -0,0 +1,27 @@
+package com.freeletics.mad.example.feature.main
+
+import com.freeletics.mad.navigator.compose.NavDestination
+import com.freeletics.mad.whetstone.AppScope
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+
+class Main
+
+@Module
+@ContributesTo(Main::class)
+object MainRetainedModule {
+    @Provides
+    fun provideNavDestinations(destinations: @JvmSuppressWildcards Set<NavDestination>): DestinationsHolder {
+        return DestinationsHolder(destinations)
+    }
+}
+
+// Helper class to inject a Set into MainUi
+data class DestinationsHolder(
+    val destinations: Set<NavDestination>,
+)
```

Comment:
You can simply inject `Set<NavDestination>` into the composable, not need for the `DestinationsHolder` wrapper.